public/app/features/dashboard-scene/scene/VizPanelHeaderActions.tsx

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useMemo } from 'react';
+
+import {
+  GroupByVariable,
+  SceneComponentProps,
+  sceneGraph,
+  SceneObjectBase,
+  SceneObjectState,
+  SceneQueryRunner,
+  VizPanel,
+  SceneDataState,
+  sceneUtils,
+} from '@grafana/scenes';
+
+import { PanelGroupByAction } from './PanelGroupByAction';
+
+export interface VizPanelHeaderActionsState extends SceneObjectState {
+  hideGroupByAction?: boolean;
+}
+
+export class VizPanelHeaderActions extends SceneObjectBase<VizPanelHeaderActionsState> {
+  static Component = VizPanelHeaderActionsRenderer;
+
+  constructor(state: Partial<VizPanelHeaderActionsState>) {
+    super({
+      hideGroupByAction: state.hideGroupByAction ?? false,
+      ...state,
+    });
+
+    this.addActivationHandler(this._onActivate);
+  }
+
+  private _onActivate = () => {
+    if (!this.parent || !(this.parent instanceof VizPanel)) {
+      throw new Error('VizPanelHeaderActions must be a child of a VizPanel');
+    }
+  };
+
+  public updatePanelShowAlwaysMenu(showMenuAlways: boolean) {
+    const panel = this.parent;
+
+    if (panel instanceof VizPanel && panel.state.showMenuAlways !== showMenuAlways) {
+      panel.setState({ showMenuAlways });
+    }
+
+    return () => {
+      if (panel instanceof VizPanel && panel.state.showMenuAlways) {
+        panel.setState({ showMenuAlways: false });
+      }
+    };
+  }
```

Comment:
I think this logic should be built into PanelChrome, so that if there are actions and menu, the menu should always be shown

Reply:
that's a good call, I didn't know if we wanted this as a general feature or not, but it does make sense.

Reply:
I've looked in this but I'm not sure it'll work in the panelchrome, mainly because the panelChrome just sees either a ReactElement or nothing, and all vizPanels in dashboards would have a actions components that may or may not render. so from the perspective of panelchrome it would always show the menu because the wrapper will always be there